aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/Translator.java

```diff
@@ -182,8 +182,16 @@ static ResourceModel translateFromReadResponse(
    * @return awsRequest the aws service request to delete a resource
    */
   static DeleteDomainRequest translateToDeleteRequest(final ResourceModel model) {
+
+    String domainName = model.getDomainName();
+
+    if (model.getArn() != null && domainName == null) {
+      Arn domainArn = ArnUtils.fromArn(model.getArn());
+      domainName = domainArn.shortId();
+    }
+
```

Comment:
Can we add a unit test for this?

Reply:
Yes, unit test added for delete handler.